src/rust/cryptography-x509/src/common.rs

```diff
@@ -17,6 +17,10 @@ impl AlgorithmIdentifier<'_> {
     pub fn oid(&self) -> &asn1::ObjectIdentifier {
         self.params.item()
     }
+
+    pub fn is_rsa_key(&self) -> bool {
+        matches!(self.params, AlgorithmParameters::Rsa(_))
+    }
```

Comment:
Wellll, now this excludes RsaPss which is a distinct key type.

I think maybe you just want an if statement in the validator :-)

Reply:
Yeah, that'll be cleaner!